library/arith.lisp

```diff
@@ -480,6 +480,99 @@ The fields are defined as follows:
       (lisp Double-Float (x y)
         (cl:coerce (cl:/ x y) 'cl:double-float)))))
 
+(coalton-toplevel
+  (define-class (Trigonometric :a)
+    (sin   (:a -> :a))
+    (cos   (:a -> :a))
+    (tan   (:a -> :a))
+    (sinh  (:a -> :a))
+    (cosh  (:a -> :a))
+    (tanh  (:a -> :a))
+    (asin  (:a -> :a))
+    (acos  (:a -> :a))
+    (atan  (:a -> :a))
+    (asinh (:a -> :a))
+    (acosh (:a -> :a))
+    (atanh (:a -> :a)))
+
+  (define-class (Exponentiable :a)
+    (**      (:a -> :a -> :a))
+    (logBase (:a -> :a -> :a)))
+
+  (declare sqrt ((Dividable Integer :a) (Exponentiable :a) => :a -> :a))
+  (define (sqrt x)
```

Comment:
can we add `sqrt` to the `exponentiable` class? I know it's technically redundant but sqrt is used an awful lot and can be specialized.

Reply:
Would it make sense to use the `specialize` form there?

Reply:
I suppose we can "just" rely on specializing here.